cnd/src/swap_protocols/rfc003/bitcoin/htlc_events.rs

```diff
@@ -85,15 +78,16 @@ impl<Bitcoin: bitcoin::Bitcoin + bitcoin::Network> HtlcRedeemed<Bitcoin, asset::
         start_of_swap: NaiveDateTime,
     ) -> anyhow::Result<Redeemed<::bitcoin::Transaction>> {
         let connector = self.clone();
-        let pattern = TransactionPattern {
-            to_address: None,
-            from_outpoint: Some(htlc_deployment.location),
-            unlock_script: Some(vec![vec![1u8]]),
-        };
 
-        let transaction = matching_transaction(connector, pattern, start_of_swap)
-            .await
-            .context("failed to find transaction to redeem from htlc")?;
+        let transaction = watch_for_transaction(
```

Comment:
How about adding a layer of abstraction on top of this function, like what we do in ethereum with `watch_for_contract_creation` -> `matching_transaction_receipt`.  That way the `unlock_script` (const) would not need to be known to `htlc_events` and the function could return the secret directly.

Reply:
I implemented it the same way as for Ethereum - the secret is extracted in `ethereum/htlc_events.rs`, not in `watch_for_event`. I actually like it that way - extracting the secret is a different concern than finding the transaction/event for me.

Reply:
Discussed this further with Thomas and understand better what you meant Tobin :)